src/Token/AccessToken.php

```diff
@@ -100,6 +100,21 @@ public function getUid()
     }
 
     /**
+     * Checks if the token has expired.
+     *
+     * @return boolean true if the token has expired, false otherwise.
+     * @throws InvalidArgumentException if the expires is not set on the token.
+     */
+    public function hasExpired()
+    {
+        if (!isset($this->expires)) {
+            throw new InvalidArgumentException('"expires" is not set on the token');
```

Comment:
I thought about this too when I was writing it and went through all descriptions for them. I figured that it's an `InvalidArgumentException` because 'expires' is an invalid argument because it's not set. Another thought was to not throw and exception at all, but return false instead if not set.